docs/developer-guide/plugins-howto.md

```diff
@@ -292,65 +385,105 @@ export default function(state = { text: 'Initial Text' }, action) {
-                .switchMap((response) => Rx.Observable.of(loadedData(response.data)))
-                .catch(e => Rx.Observable.of(loadError(e.message)));
+            return Observable.defer(() =>
+                axios.get("version.txt")
+            )
+                .switchMap((response) =>
+                    Observable.of(
+                        loadedData(response.data)
+                    )
+                )
+                .catch(e =>
+                    Observable.of(
+                        loadError(e.message)
+                    )
+                );
         });
-};
 
 export default {
     loadDataEpic
 };
 ```
 
-### js/plugins/SideEffectComponent.jsx
+### js/plugins/Sample.jsx
 
 ```javascript
-import React from 'react';
-import { connect } from 'react-redux';
-import PropTypes from 'prop-types';
-import {get} from 'lodash';
-
-import { loadData } from '../actions/sample';
-import sampleEpics from '../epics/sample';
-import sample from '../reducers/sample';
+import React from "react";
+import { connect } from "react-redux";
+import PropTypes from "prop-types";
+
+import { createPlugin } from "@mapstore/utils/PluginsUtils";
+
+import { loadData } from "@js/actions/sample";
+import sampleEpics from "@js/epics/sample";
+import sample from "@js/reducers/sample";
+
+const SideEffectComponent = ({
+    style,
+    text,
+    onLoad
+}) => {
+    return (
+        <div style={style}>
+            Text: {text}
+            <button onClick={() => onLoad()}>
+                Load
+            </button>
+        </div>
+    );
+};
 
-class SideEffectComponent extends React.Component {
-    static propTypes = {
-        text: PropTypes.string,
-        onLoad: PropTypes.func
-    };
+SideEffectComponent.propTypes = {
+    style: PropTypes.object,
+    text: PropTypes.string,
+    onLoad: PropTypes.func
+};
 
-    render() {
-        const style = { position: "absolute", top: "100px", left: "100px", zIndex: 1000000 };
-        return <div style={style}>Text: {this.props.text} <button onClick={this.props.onLoad}>Load</button></div >;
-    }
-}
+SideEffectComponent.defaultProps = {
+    style: {
+        position: "absolute",
+        top: 100,
+        left: 100,
+        zIndex: 2000
+    },
+    text: "",
+    onLoad: () => {}
+};
 
-const ConnectedSideEffect = connect((state) => {
+const Sample = connect((state) => {
     return {
-        text: get(state, 'sample.text')
+        text: state?.sample?.text
     };
 }, {
-        onLoad: loadData // connected action
-    })(SideEffectComponent);
-
-export const SideEffectPlugin = ConnectedSideEffect;
-export const reducers = {sample};
-export const epics = sampleEpics;
+    // connected action
+    onLoad: loadData
+})(SideEffectComponent);
+
+export default createPlugin("Sample", {
+    component: Sample,
+    reducers: {
+        sample
+    },
+    epics: sampleEpics
+});
 ```
 
 ## Plugins that are containers of other plugins
```

Comment:
`Plugins that are containers of other plugins` is mad. Use "Plugin Containers", or "Injecting plugin rendering"